### PR TITLE
Support keypairs

### PR DESCRIPTION
**Pull request by maxogden**
Comments inline

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -107,12 +132,15 @@ module.exports = function townshipToken (maindb, options) {
...
+    if (algorithm) options.algorithm = algorithm
+    var secretOrPublicKey = options.secret || secret
+    if (publicKey) secretOrPublicKey = publicKey
+
     db.createReadStream().pipe(through.obj(each, end))
 
     function each (token, enc, next) {
       try {
-        jwt.verify(token.key, secret)
+        jwt.verify(token.key, secretOrPublicKey, options)
```
Comment (maxogden) [inline index.js]: note: options here was also previously unused

File: test.js [hunk 2]
```diff
@@ -114,3 +159,28 @@ test('clean up invalid tokens list', function (t) {
     }, 1500)
   })
 })
+
+test('clean up invalid tokens list with keys', function (t) {
```
Comment (maxogden) [inline test.js]: hope its ok to repeat tests like this

File: index.js [hunk 3]
```diff
@@ -4,21 +4,42 @@ var sublevel = require('subleveldown')
...
+* })
 **/
+
 module.exports = function townshipToken (maindb, options) {
   options = options || {}
-  var secret = options.secret || 'not a secret'
+  var secret = options.secret
+  var publicKey = options.publicKey
+  var privateKey = options.privateKey
+  var algorithm = options.algorithm
```
Comment (maxogden) [inline index.js]: annoying that `jsonwebtoken` doesnt allow you to pass in an options object, but rather makes you do the `secretOrPublicOrPrivateKey` magic string thing. so we have to do lots of options checking to wrap their api

**Comment by sethvincent** [comment]
thanks!

**Pull request merged**